### PR TITLE
[CHORE] Refactor dependencies, update scalacheck

### PR DESCRIPTION
**Pull request by mmrozek**
# Description

- Move dependencies to separate file
- update scalacheck

CAUTION: a majority of changes is caused by renaming of `PropertyChecks` trait and formatting all touched files. Please don't be afraid ;)

### CODE REVIEW

File: repo.nix [hunk 1]
```diff
@@ -5,10 +5,119 @@
       "sbtVersion" = "1.2.8";
     }];
   "repos" = {
+    "nix-local-preloaded" = "";
     "nix-public" = "";
     "nix-rocksDb" = "";
   };
   "artifacts" = {
+    "nix-local-preloaded/file:/Users/mmrozek/.sbtix/preloaded/com/trueaccord/lenses/lenses_2.12/0.4.12/lenses_2.12-0.4.12.jar" = {
```
Comment (mirkoAlic) [inline repo.nix:13]: =/

File: .scalafmt.conf [hunk 2]
```diff
@@ -1,3 +1,3 @@
-version = "2.3.0"
+version = "2.7.1"
```
Comment (ntallar) [inline .scalafmt.conf:1]: What do you think about including scalafmt checks on circleci?

Comment (mmrozek) [reply .scalafmt.conf:1]: At some point, we need to do that. But I prefer to do that in separate PR

**Review by kapke — APPROVED** [review]
One minor thing and look good to me!

**Pull request merged**